biogeochem/EDCohortDynamicsMod.F90

```diff
@@ -150,318 +145,144 @@ Module EDCohortDynamicsMod
-
-    if (new_cohort%dbh <= 0._r8 .or. new_cohort%n == 0._r8 .or. new_cohort%pft == 0 ) then
-       write(fates_log(),*) 'ED: something is zero in create_cohort', &
-                             new_cohort%dbh,new_cohort%n, &
-                             new_cohort%pft
-       call endrun(msg=errMsg(sourcefile, __LINE__))
-    endif
-
-    ! Assign canopy extent and depth
-    if(hlm_use_sp.eq.ifalse)then
-       call carea_allom(new_cohort%dbh,new_cohort%n,spread,new_cohort%pft, &
-            new_cohort%crowndamage,new_cohort%c_area)
-    else
-      new_cohort%c_area = carea ! set this from previously precision-controlled value in SP mode
-    endif
-    ! Query PARTEH for the leaf carbon [kg]
-    leaf_c = new_cohort%prt%GetState(leaf_organ,carbon12_element)
-
-    new_cohort%treelai = tree_lai(leaf_c, new_cohort%pft, new_cohort%c_area,    &
-                                  new_cohort%n, new_cohort%canopy_layer,               &
-                                  patchptr%canopy_layer_tlai,new_cohort%vcmax25top )
-
-    if(hlm_use_sp.eq.ifalse)then
-       new_cohort%treesai = tree_sai(new_cohort%pft, new_cohort%dbh, &
-            new_cohort%crowndamage, new_cohort%canopy_trim,   &
-            new_cohort%c_area, new_cohort%n, new_cohort%canopy_layer, &
-            patchptr%canopy_layer_tlai, new_cohort%treelai,new_cohort%vcmax25top,2 )
-    end if
-
-
-    ! Put cohort at the right place in the linked list
-    storebigcohort   => patchptr%tallest
-    storesmallcohort => patchptr%shortest
-
-    if (associated(patchptr%tallest)) then
-       tnull = 0
-    else
-       tnull = 1
-       patchptr%tallest => new_cohort
-    endif
-
-    if (associated(patchptr%shortest)) then
-       snull = 0
-    else
-       snull = 1
-       patchptr%shortest => new_cohort
-    endif
-
-    ! Allocate running mean functions
-
-    !  (Keeping as an example)
-    !! allocate(new_cohort%tveg_lpa)
-    !! call new_cohort%tveg_lpa%InitRMean(ema_lpa,init_value=patchptr%tveg_lpa%GetMean())
-
-    call InitPRTBoundaryConditions(new_cohort)
-    
-    
-    ! Recuits do not have mortality rates, nor have they moved any
-    ! carbon when they are created.  They will bias our statistics
-    ! until they have experienced a full day.  We need a newly recruited flag.
-    ! This flag will be set to false after it has experienced
-    ! growth, disturbance and mortality.
-    new_cohort%isnew = .true.
-
-    if( hlm_use_planthydro.eq.itrue ) then
-
-       nlevrhiz = currentSite%si_hydr%nlevrhiz
-
-       ! This allocates array spaces
-       call InitHydrCohort(currentSite,new_cohort)
-
-       ! zero out the water balance error
-       new_cohort%co_hydr%errh2o = 0._r8
-
-       ! This calculates node heights
-       call UpdatePlantHydrNodes(new_cohort,new_cohort%pft, &
-                                new_cohort%hite,currentSite%si_hydr)
+subroutine create_cohort(currentSite, patchptr, pft, nn, hite, coage, dbh,   &
```

Comment:
Are there instances where the old "create_cohort()" routine is still required?  I was surprised to see this still here.

Reply:
the `create_cohort` is still used to set up a cohort on a patch (think of it as a patch-level method, though I haven't moved it to be a type-bound patch method yet). the `cohort%Create` does the initing and setting of values, and is called within `create_cohort`